spec/models/miq_provision_workflow_spec.rb

```diff
@@ -53,20 +61,77 @@
+            "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test'},
+            nil, nil, {'abc' => 'tr|ue', 'blah' => 'nah'}, nil, nil)
+
+          expect(request.options[:ws_values]).to include(:abc => "tr|ue")
+        end
+
+        it "should set values when extra '|' are passed in for multiple values" do
+          request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
+            "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test'},
+            nil, nil, {'abc' => 'tr|ue', 'blah' => 'na|h'}, nil, nil)
+
+          expect(request.options[:ws_values]).to include(:blah => "na|h")
+        end
+
         it "should set values" do
           request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
-            "1.1", admin, "name=template", "vm_name=spec_test",
-            nil, nil, "abc=true", nil, nil)
+            "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test'},
+            nil, nil, {'abc' => 'true', 'blah' => 'nah'}, nil, nil)
 
           expect(request.options[:ws_values]).to include(:abc => "true")
         end
+
+        it "should set values correctly when a string is passed" do
+          Vmdb::Deprecation.silenced do
+            request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
+              "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test'},
+              nil, nil, "abc=true|blah=nah", nil, nil)
+
+            expect(request.options[:ws_values]).to include(:abc => "true")
+          end
+        end
+
+        it "should set values when only a single key value pair is passed in as a string" do
+          Vmdb::Deprecation.silenced do
+            request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
+              "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test'},
+              nil, nil, "abc=true", nil, nil)
+
+            expect(request.options[:ws_values]).to include(:abc => "true")
+          end
+        end
+
+        it "should set values when all args are passed in as a string" do
+          Vmdb::Deprecation.silenced do
+            request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
+              "1.1", admin, "name=template", "vm_name=spec_test",
+              nil, nil, "abc=true", nil, nil)
+
+            expect(request.options[:ws_values]).to include(:abc => "true")
+          end
+        end
+
+        it "should set values when only a single key value pair is passed in" do
+          Vmdb::Deprecation.silenced do
+            request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
+              "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test'},
+              nil, nil, {'abc' => 'true'}, nil, nil)
+
+            expect(request.options[:ws_values]).to include(:abc => "true")
+          end
```

Comment:
I am thinking we can get away without this tests as well.  You are already parsing single key/value pairs in other tests here.  If we do end up keeping it note that the `Vmdb::Deprecation.silenced` is not needed since there are no strings parameters being passed.